File: it/expected-structures/default_naming/default/js/lower_camel.js

export class Value {
  constructor(foo_bar) {
    this.foo_bar = foo_bar;
  }

  static decode(data) {
    const v_foo_bar = data["fooBar"];

    if (v_foo_bar === null || v_foo_bar === undefined) {
      throw new Error("fooBar" + ": required field");
    }

    return new Value(v_foo_bar);
  }

  encode() {
    const data = {};

    if (this.foo_bar === null || this.foo_bar === undefined) {
      throw new Error("fooBar: is a required field");
    }

    data["fooBar"] = this.foo_bar;

    return data;
  }
}

File: it/default_naming/structures/default/python/lower_camel.py
class Value:
  def __init__(self, _foo_bar):
    self._foo_bar = _foo_bar

  @property
  def foo_bar(self):
    return self._foo_bar

  @staticmethod
  def decode(data):
    f_foo_bar = data["fooBar"]

    if not isinstance(f_foo_bar, unicode):
      raise Exception("not a string")

    return Value(f_foo_bar)

  def encode(self):
    data = dict()

    if self._foo_bar is None:
      raise Exception("fooBar: is a required field")

    data["fooBar"] = self._foo_bar

    return data

  def __repr__(self):
    return "<Value foo_bar:{!r}>".format(self._foo_bar)

